src/main/groovy/edu/oregonstate/mist/locations/db/DiningDAO.groovy

```diff
@@ -0,0 +1,164 @@
+package edu.oregonstate.mist.locations.db
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import edu.oregonstate.mist.locations.LocationUtil
+import edu.oregonstate.mist.locations.core.DiningLocation
+import edu.oregonstate.mist.locations.core.DayOpenHours
+import net.fortuna.ical4j.data.CalendarBuilder
+import net.fortuna.ical4j.filter.Filter
+import net.fortuna.ical4j.filter.PeriodRule
+import net.fortuna.ical4j.model.Component
+import net.fortuna.ical4j.model.Dur
+import net.fortuna.ical4j.model.Period
+import net.fortuna.ical4j.model.Property
+import net.fortuna.ical4j.model.PropertyList
+import net.fortuna.ical4j.model.property.ExDate
+import org.joda.time.DateTime
+import org.joda.time.LocalDate
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * The Dining data comes from google calendar
+ */
+public class DiningDAO {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DiningDAO.class)
+
+    /**
+     * Url of Dining API. JSON formatted content that includes lat/long,
+     * name, calendar id, and today's hours for dining locations.
+     */
+    private final String uhdsURL
+
+    /**
+     * Url format of ical calendar that includes open hours for dining
+     */
+    private final String icalURL
+
+    /**
+     * Filename to store the uhdsURL cache data
+     */
+    private final String diningJsonOut
+
+    private final LocationUtil locationUtil
+
+    public DiningDAO(Map<String, String> locationConfiguration, LocationUtil locationUtil) {
+        uhdsURL = locationConfiguration.get("uhdsUrl")
+        icalURL = locationConfiguration.get("icalUrl")
+        diningJsonOut = locationConfiguration.get("diningJsonOut")
+        this.locationUtil = locationUtil
+    }
+
+    List<DiningLocation> getDiningLocations() {
+        String diningData = getDiningLocationList()
+
+        ObjectMapper mapper = new ObjectMapper(); // can reuse, share globally
+        List<DiningLocation> diners = mapper.readValue(diningData, new TypeReference<List<DiningLocation>>(){})
+        diners.unique(true) // the json datasource lists the location multiple time if it's open twice a day
+
+        diners.each {
+            def icalURL = icalURL.replace("calendar-id", "${it.calendarId}")
+            def icalFileName = it.calendarId + ".ics"
+            LOGGER.debug(icalURL)
+
+            String icalData = getIcalData(icalURL, icalFileName)
+            it.openHours = parseDiningICAL(icalData)
+        }
+        //@todo: how to deal with html in title?
+
+        //@todo: need a flag to know if it's the first time in the day taht we have flagged
+
+        diners
+    }
+
+    private String getIcalData(String icalURL, String icalFileName) {
+        // @todo: what if it's not new, but the open hours in the calendar are different for next week?
+        locationUtil.getDataFromUrlOrCache(icalURL, icalFileName)
```

Comment:
This todo and a couple of others are about efficiency. They would save off 20 seconds on each run. Since this API is not exposed to consumers, time improvements don't add a lot of value. These todos will be handled in future revisions.